### PR TITLE
feat(codeowners): List view UI and edit IssueOwners

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective:
There is a new UI for the Issue Owners page. This PR implements the List View and the Edit Ownership Rules Modal with full functionality.

## UI
_List View:_
localhost_8000_settings_sentry_projects_earth-vt_ownership_

_Edit Modal:_
localhost_8000_settings_sentry_projects_earth-vt_ownership_ (1)

### CODE REVIEW

**Comment by MeredithAnya** [comment]
do we have any tests for the new modal?

**Comment by MeredithAnya** [comment]
@NisanthanNanthakumar testing this out, when you click "Save Changes" it doesn't close the modal. Also, it lets you put a cursor in the "read only" part for the ownership rules which is pretty confusing cause then you can't type. 
Screen Shot 2021-04-07 at 11 01 12 AM

File: src/sentry/static/sentry/app/views/settings/project/projectOwnership/editRulesModal.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      <React.Fragment>
+        <Block>
+          {t('Rules follow the pattern: ')} <code>type:glob owner owner</code>
+        </Block>
+        <Block>
+          {t('Owners can be team identifiers starting with #, or user emails')}
+        </Block>
+        <Block>
+          {t('Globbing Syntax:')}
+          <CodeBlock>{'* matches everything\n? mathes any single character'}</CodeBlock>
```
Comment (MeredithAnya) [inline src/sentry/static/sentry/app/views/settings/project/projectOwnership/editRulesModal.tsx:25]: `mathes any single character` --> `matches any single character`

File: src/sentry/static/sentry/app/views/settings/project/projectOwnership/codeowners.tsx [hunk 2]
```diff
@@ -31,159 +26,58 @@ class CodeOwners extends AsyncComponent<Props, State> {
...
+            type="codeowners"
+            raw={raw}
+            dateUpdated={dateUpdated}
+            provider={provider}
+            repoName={repoName}
+            readOnly
+            controls={[
+              <Confirm
+                onConfirm={() => this.handleDelete(codeowner)}
+                message={t('Are you sure you want to remove this CodeOwners?')}
```
Comment (MeredithAnya) [inline src/sentry/static/sentry/app/views/settings/project/projectOwnership/codeowners.tsx:70]: I feel like it should either be `CODEOWNER file` or `code owner file`

**Comment by NisanthanNanthakumar** [comment]
> do we have any tests for the new modal?

@MeredithAnya the core components (Input, Modal) have their own tests. I added a new test for checking if the modal opens.

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**